plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/decoders/TimestampDecoder.java

```diff
@@ -68,16 +76,18 @@ public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
             LocalDateTime timestamp;
             if (value instanceof String) {
                 String valueString = (String) value;
-                Long epochMillis = Longs.tryParse(valueString);
-                if (epochMillis != null) {
-                    timestamp = LocalDateTime.ofInstant(Instant.ofEpochMilli(epochMillis), UTC);
+                try {
+                    timestamp = LocalDateTime.from(formatter.parse(valueString));
                 }
-                else {
+                catch (DateTimeParseException | IllegalArgumentException e) {
+                    // Compatible for Elasticsearch6
+                    // Docvalue_fields query for Elasticsearch6 will always return ISO_DATE_TIME format
                     timestamp = ISO_DATE_TIME.parse(valueString, LocalDateTime::from);
                 }
             }
             else if (value instanceof Number) {
-                timestamp = LocalDateTime.ofInstant(Instant.ofEpochMilli(((Number) value).longValue()), UTC);
+                String valueString = String.valueOf(value);
+                timestamp = LocalDateTime.from(formatter.parse(valueString));
```

Comment:
I think we could inline valueString here.
```suggestion
                timestamp = LocalDateTime.from(formatter.parse(String.valueOf(value)));
```